controllers/flinkcluster/flinkcluster_updater.go

```diff
@@ -609,11 +609,8 @@ func (updater *ClusterStatusUpdater) deriveJobStatus() *v1beta1.JobStatus {
 
 	if observedSubmitter.job != nil {
 		newJob.SubmitterName = observedSubmitter.job.Name
-		exitCode, reason := updater.deriveJobSubmitterExitCodeAndReason(observed.flinkJobSubmitter.pod)
+		exitCode, _ := updater.deriveJobSubmitterExitCodeAndReason(observed.flinkJobSubmitter.pod)
```

Comment:
The failure reason is set only for the case when the flink job actually fails. It is not set for scenarios where the flink job succeeds but the submitter fails with a non-zero exit code.

Reply:
I think we need something like this:
```
if isNonZeroExitCode(newJob.SubmitterExitCode) && len(newJob.FailureReasons) == 0 {
  newJob.FailureReasons = append(newJob.FailureReasons, reason)
}
```

in [this block](https://github.com/spotify/flink-on-k8s-operator/blob/master/controllers/flinkcluster/flinkcluster_updater.go#L719) to complete. Wdyt?